src/core/TChem_AerosolModelData.cpp

```diff
@@ -136,24 +138,163 @@ int AerosolModelData::initChem(YAML::Node &root,
             //
             simpol_info_at.gas_sp_index = it_gas->second;
            } else {
-            printf("Error : species does not exit %s in reactants of SIMPOL_PHASE_TRANSFER\n", gas_sp);
+            printf("[AerosolModelData::initChem] Error : species does not exit %s in reactants of SIMPOL_PHASE_TRANSFER\n", gas_sp);
             TCHEM_CHECK_ERROR(true,"Yaml : Error when interpreting aerosol model" );
           }
           simpol_info.push_back(simpol_info_at);
-         } else {
-        printf("Warning : TChem (amd) did not parse this reaction type %s \n", reaction_type.c_str());
-       }// ireac
-      }
-    }// item loop
-
-    } // item
-    printf("Done with reactions/phase trans...\n");
+         } // if SIMPOL_PHASE_TRANSFER 
+         else {
+          printf("[AerosolModelData::initChem] Warning : TChem (amd) did not parse this reaction type %s \n", reaction_type.c_str());
+       }
+      } // ireac loop
+    } // if MECHANISM
+
+    // Extract attributes for aerosol water sub model
+    if (type=="SUB_MODEL_ZSR_AEROSOL_WATER"){
```

Comment:
Do we get the type from the input file? If so, make sure to delete ZSR from this type.

Reply:
Yes, this type comes from the input file. To clarify here, should we change `"SUB_MODEL_ZSR_AEROSOL_WATER"` to `"SUB_MODEL_AEROSOL_WATER"` in the input file and associated code entries?

Reply:
what is `ZSR`? is the name of this model?

Reply:
The descriptor comes from the CAMP test case. ZSR stands for Zdanovskii, Stokes, and Robinson, which developed the method for approximating the aerosol water content in the manner that we use.

Reply:
It makes sense. We can keep this name. I would recommend adding a comment with the model's full name.